### PR TITLE
[move-mutator] break/continue replacement operator

### PR DESCRIPTION
**Pull request by asmie**
### Description

This PR adds new mutation operator - break/continue replacement.

This operator replaces break and continue operators with each other or with the empty statement `{ }`.

### Test Plan
Unit tests added to the new operator module.
New test asset added to tests exactly `breakcontinue` operator.

### CODE REVIEW

File: third_party/move/tools/move-mutator/src/operators/break_continue.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+use move_command_line_common::files::FileHash;
+use move_compiler::parser::ast;
+use move_compiler::parser::ast::Exp_;
+use std::fmt;
+use std::fmt::Debug;
+
+/// Break and continue mutation operator.
+/// Replaces break and continue statements with each other or deletes them.
+#[derive(Debug, Clone)]
+pub struct BreakContinue {
```
Comment (Rqnsom) [inline third_party/move/tools/move-mutator/src/operators/break_continue.rs:12]: Just sharing an idea I'm not sure I even like:
```suggestion
pub struct LoopCtrlStatement {
```

File: third_party/move/tools/move-mutator/src/operators/break_continue.rs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    operation: ast::Exp,
+}
+
+impl BreakContinue {
+    pub fn new(operation: ast::Exp) -> Self {
```
Comment (Rqnsom) [inline third_party/move/tools/move-mutator/src/operators/break_continue.rs:19]: Shall we put `#[must_use]` here? Is that common in this repo?

**Review by Rqnsom — APPROVED** [review]
LGTM

An idea:
Maybe it would be good to have example mutant tests:
A test that generates e.g. two mutants from file XXX
and then expects to find XXX_1.mutant and XXX_2.mutant

File: third_party/move/tools/move-mutator/src/operators/break_continue.rs (hunk 1 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/operators/break_continue.rs:12]: I've used a name similar to the names that are in similar tools like SuMo tool for Solidity:
https://github.com/MorenaBarboni/SuMo-SOlidity-MUtator/blob/master/src/operators/standard/break-continue-replacement.js

File: third_party/move/tools/move-mutator/src/operators/break_continue.rs (hunk 2 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/operators/break_continue.rs:19]: Right, added.

**Review by Rqnsom — APPROVED** [review]

**Comment by asmie** [comment]
Test idea added to the project backlog.
Thanks!

**Pull request merged**